src/coreclr/gc/gc.cpp

```diff
@@ -6683,6 +6688,9 @@ void gc_heap::gc_thread_function ()
 
             if (!should_proceed_with_gc())
             {
+#ifdef USE_REGIONS
+                GCToEEInterface::DiagUpdateGenerationBounds();
```

Comment:
why are we calling these for regions instead of calling AddNewRegion in `extend_soh_for_no_gc` like we do in `soh_try_fit`?